### PR TITLE
added kuskClient create staticroute and expanded respective controller

### PR DESCRIPTION
**Pull request by jasmingacic**
First swing at it.

Still missing testing and unit mock tests. 

Signed-off-by: Jasmin Gacic <jasmin.gacic@gmail.com>

### CODE REVIEW

**Review by povilasv — APPROVED** [review]

**Comment by mbana** [comment]
Will (partially?) address #45.

File: server/go/api_apis.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 
 // ApisApiController binds http requests to an api service and writes the service results to the http response
 type ApisApiController struct {
-	service      ApisApiServicer
```
Comment (jasmingacic) [inline server/go/api_apis.go:22]: go fmt ./... will have this aligned correctly no need for so many changes

File: server/go/api_create_new_static_route_service_impl.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	return &CreateNewStaticRouteApiService{kuskClient: kuskClient}
+}
+
+// CreateStaticRoute - create new static route
+func (s *CreateNewStaticRouteApiService) CreateStaticRoute(ctx context.Context, staticRouteItem StaticRouteItem) (ImplResponse, error) {
+	staticRoute, err := s.kuskClient.CreateStaticRoute(staticRouteItem.Namespace, staticRouteItem.Name, staticRouteItem.EnvoyFleetNamespace, staticRouteItem.EnvoyFleetName)
+	if err != nil {
+		return Response(http.StatusInternalServerError, err), err
+	}
+	return Response(http.StatusOK, staticRoute), nil
```
Comment (jasmingacic) [inline server/go/api_create_new_static_route_service_impl.go:37]: expected response for staticroutes POST is https://github.com/kubeshop/kuskgateway-api-server/blob/create_staticroute/api/openapi.yaml#L342 
Please transform CRD into API object

File: server/go/api_create_new_static_route_service_impl.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+// NewCreateNewStaticRouteApiService creates a default api service
+func NewCreateNewStaticRouteApiService(kuskClient kusk.Client) CreateNewStaticRouteApiServicer {
+	return &CreateNewStaticRouteApiService{kuskClient: kuskClient}
+}
+
+// CreateStaticRoute - create new static route
+func (s *CreateNewStaticRouteApiService) CreateStaticRoute(ctx context.Context, staticRouteItem StaticRouteItem) (ImplResponse, error) {
+	staticRoute, err := s.kuskClient.CreateStaticRoute(staticRouteItem.Namespace, staticRouteItem.Name, staticRouteItem.EnvoyFleetNamespace, staticRouteItem.EnvoyFleetName)
```
Comment (jasmingacic) [inline server/go/api_create_new_static_route_service_impl.go:33]: Why aren't we passing other parameters like: Paths and Hosts?
See https://github.com/kubeshop/kusk-gateway/blob/main/api/v1alpha1/staticroute_types.go#L39-L55

File: server/go/api_namespaces.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
+/*
+ * Kusk Gateway API
+ *
+ * This is the Kusk Gateway Management API
+ *
+ * API version: 1.0.0
+ * Generated by: OpenAPI Generator (https://openapi-generator.tech)
+ */
+
+package openapi
```
Comment (jasmingacic) [inline server/go/api_namespaces.go:10]: Once rebased this will no longer look like this

File: server/kusk/client_test.go [hunk 5]
```diff
@@ -135,6 +163,18 @@ func getFakeClient() client.Client {
 				Namespace: "default",
 			},
 		},
+		&kuskv1.StaticRoute{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "static-route-1",
+				Namespace: "default",
+			},
+			Spec: kuskv1.StaticRouteSpec{
```
Comment (jasmingacic) [inline server/kusk/client_test.go:171]: What happens with Paths and Hosts?

**Review by jasmingacic — COMMENTED** [review]
I merged #43 so you will have to rebase this. 

Keep in mind that namespaces controller is already implemented  in #43 so you don't overwrite it

Please make sure you run go fmt ./... to avoid having this many changes. It is making PRs hard to review

File: server/go/api_apis.go (hunk 1 above)
Comment (mbana) [reply server/go/api_apis.go]: Resolved. Did `make format`.

Please mark as resolved, if you are happy with it.

File: server/go/api_create_new_static_route_service_impl.go (hunk 2 above)
Comment (mbana) [reply server/go/api_create_new_static_route_service_impl.go]: Resolved.

Convert to `StaticRouteItem`.

Please mark as resolved, if you are happy with it.

**Review by jasmingacic — COMMENTED** [review]
Approved (I can't approve it as PR owner)

**Pull request merged**